app.py

```diff
@@ -8,157 +10,216 @@
     return rs
 
 
-@app.route('/')
+@app.route("/")
 def index():
     return _index()
 
 
 def captcha_check():
     try:
-        oform = {
-            #"privatekey": app.config['RECAPTCHA_PRIVATEKEY'],
-            "secret": app.config['RECAPTCHA_PRIVATEKEY'],
-            "remoteip": request.remote_addr,
-            #"challenge": request.form.get('recaptcha_challenge_field',['']),
-            #"response": request.form.get('recaptcha_response_field',[''])
-            "response": request.form.get('g-recaptcha-response',[''])
+        payload = {
+            "response": request.form.get("g-recaptcha-response", [""]),
+            "secret": app.config["RECAPTCHA_PRIVATEKEY"],
         }
-
-        #f = urllib.urlopen("http://api-verify.recaptcha.net/verify", urllib.urlencode(oform))
-        f = urllib.urlopen("https://www.google.com/recaptcha/api/siteverify", urllib.urlencode(oform))
-
-        #result = f.readline().replace("\n","")
-        #error = f.readline().replace("\n","")
-        d = json.load(f)
-        result = d["success"]
-        f.close()
-
-        if not result:#  != 'true':
-            #if error != 'incorrect-captcha-sol':
-            app.logger.info("ReCaptcha fail: %r, %r", oform, d)
-            #g.recaptcha_args += "&error=" + error
+        response = requests.post(
+            "https://www.google.com/recaptcha/api/siteverify", payload
+        )
+        # result = f.readline().replace("\n","")
+        # error = f.readline().replace("\n","")
+        response_text = json.loads(response.text)
+        result = response_text["success"]
+
+        if not result:  #  != 'true':
+            # if error != 'incorrect-captcha-sol':
+            app.logger.info("ReCaptcha fail: %r, %r", payload, result)
+            # g.recaptcha_args += "&error=" + error
             return False
 
     except:
-        #g.recaptcha_args += "&error=unknown"
+        # g.recaptcha_args += "&error=unknown"
         return False
     return True
 
-@app.route('/haxx', methods=["POST"])
+
+@app.route("/haxx", methods=["POST"])
 def haxx():
-    OUI_LIST = [i.decode('hex') for i in open(os.path.join(app.root_path, 'oui_list.txt')).read().split("\n") if len(i)==6]
-    g.recaptcha_args = 'k=%s' % app.config['RECAPTCHA_PUBLICKEY']
+
+    OUI_LIST = [
+        binascii.unhexlify(i)
+        for i in open(os.path.join(app.root_path, "oui_list.txt")).read().split("\n")
+        if len(i) == 6
+    ]
+    g.recaptcha_args = "k=%s" % app.config["RECAPTCHA_PUBLICKEY"]
     dt = datetime.utcnow() - timedelta(1)
-    delta = (dt - datetime(2000, 1, 1))
+    delta = dt - datetime(2000, 1, 1)
     timestamp = delta.days * 86400 + delta.seconds
 
     try:
-        mac = ''.join([chr(int(request.form[i],16)) for i in "abcdef"])
-        template = TEMPLATES[request.form['region']]
-        bundle = 'bundle' in request.form
+        mac = "".join([chr(int(request.form[i], 16)) for i in "abcdef"])
```

Comment:
This is wrong, mac needs to be `bytes`, not `str`. You are encoding the MAC address as Unicode characters here (chr means different things in py2 and py3).

Reply:
I think just `bytes(int(request.form[i], 16) for i in "abcdef")` should do the right thing.

Reply:
`bytes(int(request.form[i], 16) for i in "abcdef")` does work.